src/routers/__tests__/StackRouter-test.js

```diff
@@ -856,6 +856,139 @@ describe('StackRouter', () => {
     ]);
   });
 
+  test('Handles the navigate action with params and nested StackRouter as a first action', () => {
+    const state = TestStackRouter.getStateForAction({
+      type: NavigationActions.NAVIGATE,
+      routeName: 'main',
+      params: {
+        code: 'test',
+        foo: 'bar',
+      },
+      action: {
+        type: NavigationActions.NAVIGATE,
+        routeName: 'profile',
+        params: {
+          id: '4',
+          code: 'test',
+          foo: 'bar',
+        },
+        action: {
+          type: NavigationActions.NAVIGATE,
+          routeName: 'list',
+          params: {
+            id: '10259959195',
+            code: 'test',
+            foo: 'bar',
+          },
+        },
+      },
+    });
+
+    expect(state && state.index).toEqual(0);
+    expect(state && state.isTransitioning).toEqual(false);
+    /* $FlowFixMe */
+    expect(state && state.routes[0].index).toEqual(0);
+    expect(state && state.routes[0].routeName).toEqual('main');
+    expect(state && state.routes[0].routes[0].index).toEqual(0);
+    expect(state && state.routes[0].routes[0].routeName).toEqual('profile');
+    expect(state && state.routes[0].routes[0].routes[0].routeName).toEqual(
+      'list'
+    );
+    // expect(state).toEqual({
+    //   index: 0,
+    //   isTransitioning: false,
+    //   routes: [
+    //     {
+    //       index: 0,
+    //       key: 'Init-id-0-41',
+    //       params: { code: 'test', foo: 'bar' },
+    //       routeName: 'main',
+    //       routes: [
+    //         {
+    //           index: 0,
+    //           key: 'Init-id-0-40',
+    //           params: { code: 'test', foo: 'bar', id: '4' },
+    //           routeName: 'profile',
+    //           routes: [
+    //             {
+    //               key: 'Init-id-0-39',
```

Comment:
I fixed this test so that keys are actually predictable, so you can actually use this commented out stuff instead